docs/guides/custom-404-page.rst

```diff
@@ -0,0 +1,27 @@
+Use custom 404 page on my project
+=================================
+
+If you want your project to use a custom page for not found pages instead of the "Maze Found" default one,
+you can put a ``404.html`` on the root of your project's output.
+
+When a 404 is returned, Read the Docs checks if there is a ``404.html`` in the root of your project's output and uses it if it exists.
+
+As the ``404.html`` will be returned for all the URLs where the real page was not found,
+all its resources URLs and links must be absolute (starting with a ``/``),
+otherwise they will not work when a user clicks on them.
+
+In case you don't want to deal with these links manually,
+or you want to use the same style for the 404 page than your theme in an easy way,
+you can use the `sphinx-notfound-page`_ extension.
```

Comment:
Are we in the point to recommend the extension? I think there was some discussion to wait to have something solid fist.

Reply:
I think that we are in that point now. We have been testing it for a while now in our own docs and there are people already using it. Also, it has tests and its own docs now.

Reply:
This is after releasing v0.4, though.